packages/db/src/indexes/basic-index.ts

```diff
@@ -0,0 +1,439 @@
+import { defaultComparator, normalizeValue } from "../utils/comparison.js"
+import { BaseIndex } from "./base-index.js"
+import type { CompareOptions } from "../query/builder/types.js"
+import type { BasicExpression } from "../query/ir.js"
+import type { IndexOperation } from "./base-index.js"
+
+/**
+ * Options for range queries
+ */
+export interface RangeQueryOptions {
+  from?: any
+  to?: any
+  fromInclusive?: boolean
+  toInclusive?: boolean
+}
+
+/**
+ * Options for Basic index
+ */
+export interface BasicIndexOptions {
+  compareFn?: (a: any, b: any) => number
+  compareOptions?: CompareOptions
+}
+
+/**
+ * Basic index using Map + sorted Array.
+ *
+ * - Map for O(1) equality lookups
+ * - Sorted Array for O(log n) range queries via binary search
+ * - O(n) updates to maintain sort order
+ *
+ * Simpler and smaller than BTreeIndex, good for read-heavy workloads.
+ * Use BTreeIndex for write-heavy workloads with large collections.
+ */
+export class BasicIndex<
```

Comment:
Suggestion: For many users it may not be clear which index they should pick based on the names `BasicIndex` and `BTreeIndex`. Perhaps we should rename them to clarify this: `BasicIndex` -> `ReadOptimizedIndex` and `BTreeIndex` -> `WriteOptimizedIndex`. The docstrings can then explain like they do now that `ReadOptimizedIndex` is backed by a Map + sorted Array and its performance implications and that `WriteOptimizedIndex` is backed by a B+ tree.